core/services/subscription.go

```diff
@@ -42,7 +42,10 @@ func StartJobSubscription(job models.JobSpec, head *models.Head, store *strpkg.S
 
 	initrs := job.InitiatorsFor(models.LogBasedChainlinkJobInitiators...)
 
-	nextHead := head.NextInt() // Exclude current block from subscription
+	nextHead := big.NewInt(0)
```

Comment:
Does this mean if there are no heads and REPLAY_FROM_BLOCK is not set it will rescan from the 0 block? If so, that probably isn't right.

Reply:
@samsondav correct. This was the request in the story. Can you comment @spooktheducks?
https://app.clubhouse.io/chainlinklabs/story/6600/add-nil-check-fallback-for-replay-from-block-when-heads-table-is-empty